unit-tests/Zephir/Test/TypeHintsTest.php

```diff
@@ -0,0 +1,114 @@
+
+    public function setUp()
+    {
+        /* Store the current directory before to be change */
+        $this->pwd = getcwd();
+    }
+
+    public function testTypeHintsZend2()
+    {
+        chdir(__DIR__ . DIRECTORY_SEPARATOR . '_files/typehints');
+        $config = new Config();
+        $logger = new Logger($config);
+        $backend = new Zend2($config);
+        $parser = new ParserManager(new Parser(), $logger);
+        $compiler = new Compiler($config, $logger, $backend, $parser);
+        $command = new CommandGenerate(new CommandsManager);
+        $compiler->generate($command);
+        $this->assertSame(
+            implode(PHP_EOL, file('ext/typehints.c', FILE_IGNORE_NEW_LINES)),
+            implode(PHP_EOL, file('expected2.c', FILE_IGNORE_NEW_LINES))
+        );
+        $this->assertSame(
+            implode(PHP_EOL, file('ext/typehints/args.zep.h', FILE_IGNORE_NEW_LINES)),
+            implode(PHP_EOL, file('expected_args2.h', FILE_IGNORE_NEW_LINES))
+        );
+        $this->assertSame(
+            implode(PHP_EOL, file('ext/typehints/retval.zep.h', FILE_IGNORE_NEW_LINES)),
+            implode(PHP_EOL, file('expected_retval2.h', FILE_IGNORE_NEW_LINES))
+        );
+        $this->assertSame(
+            implode(PHP_EOL, file('ext/typehints/both.zep.h', FILE_IGNORE_NEW_LINES)),
+            implode(PHP_EOL, file('expected_both2.h', FILE_IGNORE_NEW_LINES))
+        );
+    }
+
+    public function testTypeHintsZend3()
+    {
+        chdir(__DIR__ . DIRECTORY_SEPARATOR . '_files/typehints');
+        $config = new Config();
+        $logger = new Logger($config);
+        $backend = new Zend3($config);
+        $parser = new ParserManager(new Parser(), $logger);
+        $compiler = new Compiler($config, $logger, $backend, $parser);
+        $command = new CommandGenerate(new CommandsManager);
+        $compiler->generate($command);
+        $this->assertSame(
+            implode(PHP_EOL, file('ext/typehints.c', FILE_IGNORE_NEW_LINES)),
+            implode(PHP_EOL, file('expected3.c', FILE_IGNORE_NEW_LINES))
+        );
+        $this->assertSame(
+            implode(PHP_EOL, file('ext/typehints/args.zep.h', FILE_IGNORE_NEW_LINES)),
+            implode(PHP_EOL, file('expected_args3.h', FILE_IGNORE_NEW_LINES))
+        );
+        $this->assertSame(
+            implode(PHP_EOL, file('ext/typehints/retval.zep.h', FILE_IGNORE_NEW_LINES)),
+            implode(PHP_EOL, file('expected_retval3.h', FILE_IGNORE_NEW_LINES))
+        );
+        $this->assertSame(
+            implode(PHP_EOL, file('ext/typehints/both.zep.h', FILE_IGNORE_NEW_LINES)),
+            implode(PHP_EOL, file('expected_both3.h', FILE_IGNORE_NEW_LINES))
+        );
+    }
+
+    /**
+     * Restore current directory, and clean config.json.
+     */
+    public function tearDown()
+    {
+        if (file_exists('.temp')) {
+            exec('rm -rf .temp/');
+        }
+
+        if (file_exists('ext')) {
+            exec('rm -rf ext/');
```

Comment:
Could you adopt this to use in AppVeyor (Windows) too

Reply:
Hmm. fair point. I'll need to take a peek at the other test I pulled this block from and adjust it, too.

Reply:
This is addressed. Now handling the other issues that are preventing this WIP from becoming a true PR.